pyccel/parser/semantic.py

```diff
@@ -446,13 +446,15 @@ def get_import(self, name):
         """."""
 
         imp = None
-
         container = self.namespace
         while container:
 
-            if name in container.imports['imports']:
-                imp =  container.imports['imports'][name]
-                break
+            #case import x as y
+            for key,_ in container.imports['imports'].items():
+                if str(key) == str(name) or (isinstance(key, AsName)
+                                            and str(key.target) == str(name)):
```

Comment:
Do you definitely need the `str` calls?
If not, I think this line can be simplified thanks to the magic method `AsName.__eq__`
See : https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L594

Reply:
I cast to str because sometimes it can be a comparison between string and symbol. a magic method should be a good thing to add.

Reply:
Hmm, maybe we need to open an issue in epic #665 to get rid of `Symbol` too